src/ws/cockpitauth.c

```diff
@@ -1045,6 +1047,16 @@ cockpit_session_create (CockpitAuth *self,
                               TRUE);
     }
 
+  lang = cockpit_web_server_parse_cookie (headers, "CockpitLang");
+  if (!lang)
+    {
+      languages = cockpit_web_server_parse_languages (headers, NULL);
+      lang = cockpit_locale_from_language (languages[0], "UTF-8", NULL);
+    }
+
+  if (lang)
+    env = g_environ_setenv (env, "LANG", lang, TRUE);
```

Comment:
See above, I believe LC_MESSAGES or LC_ALL would be more correct here. The former if it's really just a language selector and the country should be unaffected and left to the server's default.